### ISSUE TITLE
Feature Request - AWS Profile Support for S3

### ISSUE THREAD

**Issue by enly1** [mkloubert/vs-deploy#16]
Hi,

Back again with more unreasonable demands.

I work in an environment that utilises multiple accounts every day and could really do with being able to specify the aws profile that should be used when deploying to S3 at a target level.

Any chance you could add support for selecting the credentials from the default aws credentials store if a profile is specified on the s3 target ?

Had a quick look at the node sdk and it looked like it would be an easy addition:

var credentials = new AWS.SharedIniFileCredentials({profile: 'work-account'});
AWS.config.credentials = credentials;

From the code, it looks like you already setup the AWS config as part of plugin initialisation, so quick property check and credentials selection if requested would probably do it.

Bottom example @ http://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-shared.html

Would just do a pull request, but node isn't my thing.

**Comment by mkloubert** [comment]
**Available since version 3.30.2 now!**

[I also have added support for all other supported credential providers that are supported by the SDK.](https://github.com/mkloubert/vs-deploy/wiki/target_s3bucket#type)

The following **example**, what is more interesting for you, **shows**, how to use a different account from the **shared ini** file:

```json
{
    "deploy": {
        "targets": [
            {
                "type": "s3bucket",
                "name": "My S3 bucket",
                "bucket": "my-bucket",

                "credentials": {
                     "type": "shared",
                     "config": {
                          "profile": "work-account"
                     }
                }
            }
        ]
    }
}
```

And the ini file looks like that:

```ini
[default]
aws_access_key_id = <DEFAULT-ACCESS-KEY>
aws_secret_access_key = <DEFAULT-SECRET-ACCESS-KEY>

[work-account]
aws_access_key_id = <YOUR-ACCESS-KEY>
aws_secret_access_key = <YOUR-SECRET-ACCESS-KEY>
```

**Comment by enly1** [comment]
:1st_place_medal: Working nicely for shared creds provider.
